File: MultiLanguageSandbox/src/test/deps/haskell/42.hs
signOfNumber :: Int -> String
signOfNumber n
    | n > 0     = "positive"
    | n < 0     = "negative"
    | otherwise = "zero"
check :: (Int -> String) -> IO ()
check signOfNumberFunc = do
  let testCases = 
        [ (95, "positive"),
          (-5, "negative"),
          (0, "zero"),
          (-1000000000, "negative"),
          (1000000000, "positive")
        ]

  mapM_ (\(number, expected) -> testCase number expected) testCases
  where
    testCase number expected = do
      let result = signOfNumberFunc number
      putStrLn $
        "Input: " ++ show number ++
        ", Expected: " ++ show expected ++ ", Result: " ++ show result ++
        if result == expected then " (Pass)" else error "Test failed. Exiting..."

main :: IO ()
main = check signOfNumber

File: MultiLanguageSandbox/src/test/deps/perl/42.pl

sub determine_sign {
    my ($N) = @_;
    if ($N > 0) {
        return 'positive';
    } elsif ($N == 0) {
        return 'zero';
    } else {
        return 'negative';
    }
}


sub check {
    my $func = shift;
    die "Test failed! Expected 'positive', got " . &$func(10) unless &$func(10) eq 'positive';
    die "Test failed! Expected 'zero', got " . &$func(0) unless &$func(0) eq 'zero';
    die "Test failed! Expected 'negative', got " . &$func(-1) unless &$func(-1) eq 'negative';
    die "Test failed! Expected 'negative', got " . &$func(-10000) unless &$func(-10000) eq 'negative';
    die "Test failed! Expected 'positive', got " . &$func(999) unless &$func(999) eq 'positive';
    print "All tests passed!\n";
}

check(\&determine_sign);